### ISSUE TITLE
Interop with ecology related packages

### ISSUE THREAD

**Issue by kescobo** [BioJulia/Microbiome.jl#7]
Continuing discussion from [richardreeve/Diversity.jl/issues/23](https://github.com/richardreeve/Diversity.jl/issues/23). @mkborregaard - we can keep having the discussion over there if you think it's better, but I thought a more general conversation about interoperability in this package would also be useful, and maybe better to get that thread off-topic.

**Comment by mkborregaard** [comment]
Interop question: from your readme:
```
df = DataFrame(species=["E. coli", "B. fragilis", "L. casei"],
               sample1=[0.1, 0.4, 0.5],
               sample2=[0.3, 0.7, 0.0],
               sample3=[0.0, 0.2, 0.8])
```
I actually had the impression that it was a universal standard to have species on the columns and samples on the rows - at least it pervades the R ecosystem (http://r4ds.had.co.nz/tidy-data.html) also for community matrices – and that's what SpatialEcology does. Is it different for microbiomics? Do you assume that's the rotation throughout your code?

**Comment by kescobo** [comment]
I do assume that, as the outputs from [our lab's tools](https://bitbucket.org/biobakery/biobakery/wiki/Home) put samples in the columns and features on rows. I don't know if that's something common for microbiome stuff generally, I think it just makes the most sense for how the data is typically handled (most things are run on a per-sample basis, and then merged prior to final analysis), but most people in lab use R for downstream analyses and I think they just transpose the matrices.

**Comment by mkborregaard** [comment]
Muarhaha, that's exactly the thing :-D There are exactly two ways of defining a species-by-site-matrix, and by necessity some will do one thing, others will do the other. I guess the possibilities are either 1) have two distinct ecosystems, 2) leave the headache to the user, or 3) agree on something. It's an interesting argument that for many types of analyses the matrix may be sliced more between sites so this should be faster, though I'm not 100% sure that makes a difference when matrices are sparse.

**Comment by mkborregaard** [comment]
I've prepared the PR, let me comment on that in detail.

**Comment by kescobo** [comment]
@mkborregaard Alright, I've merged my most recent development branch into master (#12). Please make your PR against the `master` branch. I'm likely to keep developing in the meantime, but I can rebase onto your changes once they're ready.

**Comment by kescobo** [comment]
Addressed by #15
